samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/util/SamigoExpressionParser.java

```diff
@@ -35,6 +35,10 @@
   public static String[] oldSamigoParserVars = {"SIN", "COS", "TAN", "ASIN", "ACOS", "ATAN", "ABS", "EXP", "SGN", "SQRT", "LOG10", "LN", "PI", "E", "SIGN", "LOG", "log"};
   public static String[] newSamigoParserVars = {"sin", "cos", "tan", "asin", "acos", "atan", "abs", "exp", "sgn", "sqrt", "log10", "ln", "pi", "e", "sgn",  "ln",  "ln"};
 
+  // Special case for factorial replacement
+  public static String OLD_FACTORIAL_PATTERN = "(?i)factorial\\((\\d+)\\)";
+  public static String NEW_FACTORIAL_PATTERN = "prod(i, 1, $1, i)";
```

Comment:
This should work. I have a minor concern that when creating a question, if somebody punches in a negative number, I expect the "NaN" validation message would show the 'prod' expression, and this might confuse users. I think suffixing "!" would be more clear: http://mathparser.org/mxparser-math-collection/operators/
But I'm fine with leaving this as 'prod'

Reply:
okay i changed the replacement pattern